terra/logger.py

```diff
@@ -623,6 +623,31 @@ def handle_warning(message, category, filename, lineno, file=None, line=None):
 logger = getLogger(__name__)
 
 
+def _checkLevel(settings):
+  '''
+  Translate ``settings.logging.level`` to integer.
+  '''
+  try:
+    log_level = settings.logging.level
+  except AttributeError:
+    return 0
+
+  if isinstance(log_level, str):
+    log_level = log_level.upper()
+  return logging._checkLevel(log_level)
+
+
+def _disableDebug(settings, package_names):
+  '''
+  Disable debug logging for specific packages when ``settings.logging.level``
+  is between ``DEBUG1`` and ``DEBUG3`` (e.g., only enable for ``DEBUG4``)
+  '''
+  log_level = _checkLevel(settings)
+  if DEBUG3 <= log_level and log_level <= DEBUG1:
+    for package_name in package_names:
+      getLogger(package_name).setLevel('INFO')
```

Comment:
```suggestion
def _disableDebug(package_names, enable_at=DEBUG4, disable_up_to=INFO):
  '''
  Disable debug logging for specific packages when
  ``terra.settings.logging.level`` is exclusively between ``enable_at`` and
  ``disable_up_to`` (e.g., only enable levels ``DEBUG1-3`` when the logging
  level is set to  ``DEBUG4``)

  Arguments
  ---------
  enable_at:
      The logger level where logging is reenabled. Default: ``DEBUG4``
  disable_up_to:
      The lowest logging level that is still enabled. Default: ``INFO``
  '''
  from terra import settings
  log_level = _checkLevel(settings.logging.level)
  if enable_at < log_level and log_level < disable_up_to:
    for package_name in package_names:
      getLogger(package_name).setLevel(disable_up_to)
```

You'll probably want to reword some of this, but that was my general idea.